model/project_ref.go

```diff
@@ -530,6 +518,171 @@ func recursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Valu
 	}
 }
 
+func setRepoFieldsFromProjects(repoRef *RepoRef, projectRefs []ProjectRef) {
+	if len(projectRefs) == 0 {
+		return
+	}
+	reflectedRepo := reflect.ValueOf(repoRef).Elem().Field(0) // specifically references the ProjectRef part of RepoRef
+	for i := 0; i < reflectedRepo.NumField(); i++ {
+		// for each field in the repo, look at each field in the project ref
+		var firstVal reflect.Value
+		allEqual := true
+		for j, pRef := range projectRefs {
+			reflectedBranchField := reflect.ValueOf(pRef).Field(i)
+			if j == 0 {
+				firstVal = reflectedBranchField
+			} else if !reflect.DeepEqual(firstVal.Interface(), reflectedBranchField.Interface()) {
+				allEqual = false
+				break
+			}
+		}
+		// if we got to the end of the loop, then all values are the same, so we can assign it to reflectedRepo
+		if allEqual {
+			reflectedRepo.Field(i).Set(firstVal)
+		}
+	}
+}
+
+func (p *ProjectRef) createNewRepoRef(u *user.DBUser) (repoRef *RepoRef, err error) {
+	repoRef = &RepoRef{ProjectRef{
+		Id:      mgobson.NewObjectId().Hex(),
+		Owner:   p.Owner,
+		Repo:    p.Repo,
+		Enabled: utility.TruePtr(),
+		Admins:  []string{},
+	}}
+
+	allEnabledProjects, err := FindMergedEnabledProjectRefsByOwnerAndRepo(p.Owner, p.Repo)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding all enabled projects")
+	}
+	// for every setting in the project ref, if all enabled projects have the same setting, then use that
+	defer func() {
+		err = recovery.HandlePanicWithError(recover(), err, "project and repo structures do not match")
+	}()
+	setRepoFieldsFromProjects(repoRef, allEnabledProjects)
```

Comment:
Not for this pr, but how do you feel about running the same validation that's on the project page before we save the repo ref? I figure when the project page is added to spruce, the validation logic that is currently embedded in the UI route will need to be put in its own function, which makes it easy to call here. If you agree, can you make a ticket and add it to that epic?

Reply:
Oops I didn't explain the reasoning. I was paranoid that using reflection to set complex structs here somehow doesn't work as expected, so validating the new config and logging if there's an error seemed like a reasonable safeguard. It might be a little too paranoid though

Reply:
Oh, I see. Personally I'm confident in the unit test, and I'm using this kind of reflection to set structs in mergeBranchAndRepoSettings which I've also validated pretty well, so I'm personally not concerned. Also there are only a few things that would really need validating (aliases and "is this thing enabled" really) which again would be validated when next editing the repo.